### PR TITLE
Edit and cancel frontend

### PR DESCRIPTION
**Pull request by mdulin2**

### CODE REVIEW

**Comment by mdulin2** [comment]
@jacobkrantz and @andrewjyang, I fixed the merge conflicts. This is ready for review

File: WebPhapp/backend/server.js [hunk 1]
```diff
@@ -139,6 +139,59 @@ app.get('/api/v1/prescriptions/:patientID', (req,res) => {
...
+    // finish takes a string message and a boolean (true if successful)
+    function finish(msg, success){
+        console.log(msg);
+        res.status(success ? 200 : 400).json(success);
+        return;
+    }
+
+    // Ensures that a filled or cancelled prescription cannot be altered.
+    if(prescription.cancelDate !== -1 || prescription.fillDates.length !== 0){
+      // Jacob? What to send here?
```
Comment (jacobkrantz) [inline WebPhapp/backend/server.js]: lets call `return finish('Attempt at cancelling fixed prescription', false);`

File: WebPhapp/backend/server.js [hunk 2]
```diff
@@ -139,6 +139,59 @@ app.get('/api/v1/prescriptions/:patientID', (req,res) => {
...
+    >>> curl 'http://localhost:5000/api/v1/prescriptions/edit' -H 'Acceptapplication/json, text/plain, /*' -H 'Content-Type: application/json;charset=utf-8' --data '{"prescriptionID": 3,"drugID":0,"quantity":1,"daysValid":0,"refills":0,"dispenserID":0}'
+
+    To be used in an axios call:
+        .post("/api/v1/prescription/edit",{
+            prescriptionID: 0,
+            ....
+        }
+*/
+
+app.post('/api/v1/prescriptions/edit',(req,res) => {
```
Comment (jacobkrantz) [inline WebPhapp/backend/server.js]: This is great, Max. I'll be able to go into this function and easily extend it to work with BC functions.

File: WebPhapp/backend/server.js [hunk 3]
```diff
@@ -219,6 +272,24 @@ app.post('/api/v1/prescriptions/add',(req,res) => {
     return finish("TODO: build prescription add to blockchain", true);
 });
 
+// An api endpoint that cancels the prescription associated with a
+// given prescription ID.
+// example: http://localhost:5000/api/v1/prescriptions/cancel/2
+app.get('/api/v1/prescriptions/cancel/:prescriptionID', (req,res) => {
```
Comment (jacobkrantz) [inline WebPhapp/backend/server.js]: Why is this a GET and not a POST?

File: WebPhapp/backend/server.js [hunk 4]
```diff
@@ -139,6 +139,59 @@ app.get('/api/v1/prescriptions/:patientID', (req,res) => {
...
+  Directly in terminal:
+    >>> curl 'http://localhost:5000/api/v1/prescriptions/edit' -H 'Acceptapplication/json, text/plain, /*' -H 'Content-Type: application/json;charset=utf-8' --data '{"prescriptionID": 3,"drugID":0,"quantity":1,"daysValid":0,"refills":0,"dispenserID":0}'
+
+    To be used in an axios call:
+        .post("/api/v1/prescription/edit",{
+            prescriptionID: 0,
+            ....
+        }
+*/
+
```
Comment (jacobkrantz) [inline WebPhapp/backend/server.js]: Nit: extra newline

**Comment by jacobkrantz** [comment]
@mdulin2 Looks good from the backend perspective. Just a few things to look at. Altogether, I'm happy- this will make things easier on me. Thanks, Max!

Feel free to toss it back my way if you make some changes, but I am fine if you merge.

File: WebPhapp/backend/server.js (hunk 3 above)
Comment (jacobkrantz) [reply WebPhapp/backend/server.js]: Actually I was looking at the wrong version. Question has changed. What happened to the stubbed route for `/api/v1/prescriptions/cancel/` in the final version of your PR?

File: WebPhapp/backend/server.js (hunk 1 above)
Comment (mdulin2) [reply WebPhapp/backend/server.js]: Thanks! Will do.

File: WebPhapp/backend/server.js (hunk 3 above)
Comment (mdulin2) [reply WebPhapp/backend/server.js]: Huh? Yeah, upon merging I must have accidentally deleted that. Good catch!

**Pull request merged**